application/app/services/dataverse/dataset_service.rb

```diff
@@ -3,12 +3,25 @@ class DatasetService
     include LoggingCommon
     include DateTimeCommon
 
+    AUTH_HEADER = 'X-Dataverse-key'
     class UnauthorizedException < Exception; end
+    class ApiKeyRequiredException < Exception; end
 
-    def initialize(dataverse_url, http_client: Common::HttpClient.new(base_url: dataverse_url), file_utils: Common::FileUtils.new)
+    def initialize(dataverse_url, http_client: Common::HttpClient.new(base_url: dataverse_url), api_key: nil, file_utils: Common::FileUtils.new)
       @dataverse_url = dataverse_url
       @http_client = http_client
       @file_utils = file_utils
+      @api_key = api_key
+    end
+
+    def create_dataset(dataverse_id, dataset_data)
+      raise ApiKeyRequiredException unless @api_key
+      url = "/api/dataverses/#{dataverse_id}/datasets"
+      response = @http_client.post(url, body: dataset_data.to_body, headers: { AUTH_HEADER => @api_key })
```

Comment:
Should we add Content-Type: application/json in the headers?

Reply:
You are right. let me add it